source/fab/tools/compiler_wrapper.py

```diff
@@ -0,0 +1,206 @@
+
+    @property
+    def flags(self) -> Flags:
+        ''':returns: the flags to be used with this tool.'''
+        return Flags(self._compiler.flags + self._flags)
+
+    @property
+    def suite(self) -> str:
+        ''':returns: the compiler suite of this tool.'''
+        return self._compiler.suite
+
+    @property
+    def openmp_flag(self) -> str:
+        '''Returns the flag to enable OpenMP.'''
+        return self._compiler.openmp_flag
+
+    @property
+    def has_syntax_only(self) -> bool:
+        ''':returns: whether this compiler supports a syntax-only feature.
+
+        :raises RuntimeError: if this function is called for a non-Fortran
+            wrapped compiler.
+        '''
+
+        if self._compiler.category == Category.FORTRAN_COMPILER:
+            return cast(FortranCompiler, self._compiler).has_syntax_only
+
+        raise RuntimeError(f"Compiler '{self._compiler.name}' has "
+                           f"no has_syntax_only.")
+
+    def set_module_output_path(self, path: Path):
+        '''Sets the output path for modules.
+
+        :params path: the path to the output directory.
+
+        :raises RuntimeError: if this function is called for a non-Fortran
+            wrapped compiler.
+        '''
+
+        if self._compiler.category != Category.FORTRAN_COMPILER:
+            raise RuntimeError(f"Compiler '{self._compiler.name}' has no "
+                               f"'set_module_output_path' function.")
+        cast(FortranCompiler, self._compiler).set_module_output_path(path)
+
+    def compile_file(self, input_file: Path,
+                     output_file: Path,
+                     openmp: bool,
+                     add_flags: Union[None, List[str]] = None,
+                     syntax_only: Optional[bool] = None):
+        # pylint: disable=too-many-arguments
+        '''Compiles a file using the wrapper compiler. It will temporarily
+        change the executable name of the wrapped compiler, and then calls
+        the original compiler (to get all its parameters)
+
+        :param input_file: the name of the input file.
+        :param output_file: the name of the output file.
+        :param openmp: if compilation should be done with OpenMP.
+        :param add_flags: additional flags for the compiler.
+        :param syntax_only: if set, the compiler will only do
+            a syntax check
+        '''
+
+        orig_compiler_name = self._compiler.exec_name
+        self._compiler.change_exec_name(self.exec_name)
+        if add_flags is None:
+            add_flags = []
+        if self._compiler.category is Category.FORTRAN_COMPILER:
+            # Mypy complains that self._compiler does not take the syntax
+            # only parameter. Since we know it's a FortranCompiler.
+            # do a cast to tell mypy that this is now a Fortran compiler
+            # (or a CompilerWrapper in case of nested CompilerWrappers,
+            # which also supports the syntax_only flag anyway)
```

Comment:
This suggests there might be a problem somewhere. Is there an issue to investigate?